pkg/operator/resource/resourcemerge/apps_ssa.go

```diff
@@ -0,0 +1,80 @@
+package resourcemerge
+
+import (
+	applyoperatorv1 "github.com/openshift/client-go/operator/applyconfigurations/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/ptr"
+)
+
+func SSAGenerationFor(generations []applyoperatorv1.GenerationStatusApplyConfiguration, resource schema.GroupResource, namespace, name string) *applyoperatorv1.GenerationStatusApplyConfiguration {
+	for i := range generations {
+		curr := &generations[i]
+		if *curr.Namespace == namespace &&
+			*curr.Name == name &&
+			*curr.Group == resource.Group &&
+			*curr.Resource == resource.Resource {
+
+			return curr
+		}
+	}
+
+	return nil
+}
+
+func SSASetGeneration(generations *[]applyoperatorv1.GenerationStatusApplyConfiguration, newGeneration applyoperatorv1.GenerationStatusApplyConfiguration) {
+	if generations == nil {
+		generations = &[]applyoperatorv1.GenerationStatusApplyConfiguration{}
+	}
+
+	existingGeneration := SSAGenerationFor(*generations, schema.GroupResource{Group: *newGeneration.Group, Resource: *newGeneration.Resource}, *newGeneration.Namespace, *newGeneration.Name)
+	if existingGeneration == nil {
+		*generations = append(*generations, newGeneration)
+		return
+	}
+
```

Comment:
Under what conditions will this value be present? Because we're starting from scratch with SSA, it seems reasonable to assume that we'll never encounter a case where this already has a value.  And if it did, the apiserver will reject the duplicate.

Reply:
That makes sense. This can be greatly simplified.

These are left overs from the previous iteration (when I was copying the generations from the OperatorStatus).